### PR TITLE
Ben P Bowling Challenge

### PR DESCRIPTION
**Pull request by benpartington**

### CODE REVIEW

File: lib/bowling.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  def rollball(pins)
+    @rolls << pins
+  end
+
+  def score
+    result = 0
+    number_of_roll = 0
+    10.times do
```
Comment (Ledders26) [inline lib/bowling.rb:14]: This assumes you must have 1 complete games, perhaps as a future dev could incorporate removing the score at any stage after a completed frame?

File: lib/bowling.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    @rolls[number_of_roll] + @rolls[number_of_roll + 1] + @rolls[number_of_roll + 2]
+  end
+  
+  def spare?(number_of_roll)
+    @rolls[number_of_roll] + @rolls[number_of_roll + 1] == 10
+  end
+
+  def spare_score(number_of_roll)
+    @rolls[number_of_roll] + @rolls[number_of_roll + 1] + @rolls[number_of_roll + 2]
+  end
```
Comment (Ledders26) [inline lib/bowling.rb:43]: Does the code handle spares in the final frame?  How does the game know whether to accept the third roll in the final frame?  Do we assume the user won't submit too many/too few throws?

File: lib/bowling.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+class Bowling
```
Comment (Ledders26) [inline lib/bowling.rb:1]: Nice short code, and good to have it all in one file.

File: spec/bowling_spec.rb [hunk 4]
```diff
@@ -0,0 +1,37 @@
+require_relative '../lib/bowling.rb'
+
+describe 'bowling' do
+  before{@game = Bowling.new}
+
+  it 'can roll a gutter game' do
+    20.times{@game.rollball(0)}
+    expect(@game.score).to eq(0)
```
Comment (Ledders26) [inline spec/bowling_spec.rb:8]: Nice to test a gutter game with the 20.times, I hand wrote the entry for each throw

File: spec/bowling_spec.rb [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    @game.rollball(10)
+    @game.rollball(3)
+    @game.rollball(4)
+    16.times{@game.rollball(0)}
+    expect(@game.score).to eq(24)
+  end
+
+  it 'can roll a perfect game' do
+    12.times{@game.rollball(10)}
+    expect(@game.score).to eq(300)
```
Comment (Ledders26) [inline spec/bowling_spec.rb:34]: A/A for gutter game, nice simple test

**Review by Ledders26 — COMMENTED** [review]
Good concise code.  Without doing further tests not 100% sure the final frame evaluates things like spares correctly...it may do, but I'm not sure how :)

Also, be nice to see more comprehensive set of tests for edge cases.
